test/integration/virtualnode/virtualnode_test.go

```diff
@@ -149,6 +161,101 @@ var _ = Describe("VirtualNode", func() {
+				ObjectMeta: metav1.ObjectMeta{
+					Name: meshName,
+				},
+				Spec: appmesh.MeshSpec{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"mesh": meshName,
+						},
+					},
+				},
+			}
+
+			By("creating a mesh resource in k8s", func() {
+				err := meshTest.Create(ctx, f, mesh)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("validating the resources in AWS", func() {
+				err := meshTest.CheckInAWS(ctx, f, mesh)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			cmNamespace := fmt.Sprintf("%s-%s", f.Options.ClusterName, utils.RandomDNS1123Label(6))
+			By(fmt.Sprintf("create cloudMap namespace %s", cmNamespace), func() {
+				resp, err := f.CloudMapClient.CreatePrivateDnsNamespaceWithContext(ctx, &servicediscovery.CreatePrivateDnsNamespaceInput{
+					Name: aws.String(cmNamespace),
+					Vpc:  aws.String(f.Options.AWSVPCID),
+				})
+				Expect(err).NotTo(HaveOccurred())
+				f.Logger.Info("created cloudMap namespace",
+					zap.String("namespace", cmNamespace),
+					zap.String("operationID", aws.StringValue(resp.OperationId)),
+				)
+				vnTest.CloudMapNameSpace = cmNamespace
+			})
+			//Allow CloudMap Namespace to sync
+			time.Sleep(30*time.Second)
+
+			vnBuilder := &manifest.VNBuilder{
+				ServiceDiscoveryType: manifest.CloudMapServiceDiscovery,
+				CloudMapNamespace: cmNamespace,
+			}
+
+			mb := &manifest.ManifestBuilder{
+				ServiceDiscoveryType: manifest.CloudMapServiceDiscovery,
+			}
+
+			By("Create a namespace and add labels", func() {
+				namespace, err := f.NSManager.AllocateNamespace(ctx, "appmeshtest")
+				Expect(err).NotTo(HaveOccurred())
+				vnBuilder.Namespace = namespace.Name
+				vnTest.Namespace = namespace
+				vnTest.Deployments = make(map[string]*appsv1.Deployment)
+				mb.Namespace = namespace.Name
+
+				oldNS := namespace.DeepCopy()
+				namespace.Labels = algorithm.MergeStringMap(map[string]string{
+					"appmesh.k8s.aws/sidecarInjectorWebhook": "enabled",
+					"mesh":                                   meshName,
+				}, namespace.Labels)
+
+				err = f.K8sClient.Patch(ctx, namespace, client.MergeFrom(oldNS))
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			vnName := fmt.Sprintf("vn-%s", utils.RandomDNS1123Label(8))
+			listeners := []appmesh.Listener{vnBuilder.BuildListener("http", 8080)}
+			backends := []types.NamespacedName{}
+			vn := vnBuilder.BuildVirtualNode(vnName, backends, listeners)
+
+			By("Creating a virtual node resource in k8s", func() {
+				err := vnTest.Create(ctx, f, vn)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By(fmt.Sprintf("create a deployment for VirtualNode"), func() {
+				dp := mb.BuildDeployment(vnName, 2, defaultAppImage, AppContainerPort, []corev1.EnvVar{})
+				err := f.K8sClient.Create(ctx, dp)
+				Expect(err).NotTo(HaveOccurred())
+				vnTest.Deployments[vnName]=dp
+			})
+
+			//Let Instances sync with CloudMap and Pod Readiness Gate go through
+			time.Sleep(30*time.Second)
```

Comment:
There's no better way to check by check-wait-retry-timeout instead of just sleep here? : (
This could end up being a flaky test when we enable it in the pipeline

Reply:
I mean that is what I initially started with but it was taking at least 15s+ and for that we had to introduce a delay before we reach out to cloudMap again. So, we had to introduce a sleep in some fashion inside the retry loop and the idea was why loop if we are sleeping anyways inside the retry loop and instead go with this.